### PR TITLE
Type-check test_cookies.py

### PR DESCRIPTION
**Pull request by florimondmanca**
More tests type-checking, refs #650.

### CODE REVIEW

File: tests/client/test_cookies.py [hunk 1]
```diff
@@ -99,21 +101,21 @@ class MockDispatch(Dispatcher):
...
         comment_url=None,
-        rest={"HttpOnly": None},
+        rest={"HttpOnly": ""},
         rfc2109=False,
     )
     cookies.set_cookie(cookie)
 
     client = Client(dispatch=MockDispatch())
-    client.cookies = cookies
+    client.cookies = cookies  # type: ignore
```
Comment (florimondmanca) [inline tests/client/test_cookies.py]: mypy doesn't yet support setters that accept arguments of a different type than what is returned by the getter. The official position is to `# type: ignore` for now. See: https://github.com/python/mypy/issues/3004

Comment (tomchristie) [reply tests/client/test_cookies.py]: Hrm. Well, it *is* a strange usage. Got me thinking that maybe we just shouldn’t be doing that in the first place...

Instead, we could enforce this style:

`client.cookies = httpx.Cookies(...)`

That’s possibly less surprising than setting the cookies to a dict, but having it return a Cookies instance.

We would probably want to hard check the type.

Obvs same would apply to the other setters too.

Thoughts?

Comment (florimondmanca) [reply tests/client/test_cookies.py]: Yup, sure, that would sound good to me. :-)

Comment (florimondmanca) [reply tests/client/test_cookies.py]: -> https://github.com/encode/httpx/pull/678

**Pull request merged**